assembl/static2/js/app/components/administration/survey/mediaForm.jsx

```diff
@@ -13,89 +17,103 @@ import {
-  const descriptionBottomPh = `${I18n.t('administration.ph.descriptionBottom')} ${selectedLocale.toUpperCase()}`;
-  const mediaLinkPh = `${I18n.t('administration.ph.mediaLink')} ${selectedLocale.toUpperCase()}`;
-  return (
-    <div className="form-container">
-      <div className="margin-xl">
-        <FormGroup>
-          <Checkbox checked={hasMedia} onChange={toggle}>
-            <Translate value="administration.announcementModule" />
-          </Checkbox>
-        </FormGroup>
-        {hasMedia
-          ? <div className="media-form">
-            <FormControlWithLabel
-              id="media-title"
-              label={titlePh}
-              required
-              type="text"
-              value={title}
-              onChange={(e) => {
-                return updateTitle(e.target.value);
-              }}
-            />
-            <FormControlWithLabel
-              componentClass="textarea"
-              id="media-description-top"
-              type="rich-text"
-              label={descriptionTopPh}
-              value={descriptionTop}
-              onChange={updateDescriptionTop}
-            />
-            <FormControlWithLabel
-              componentClass="textarea"
-              id="media-description-bottom"
-              type="rich-text"
-              label={descriptionBottomPh}
-              value={descriptionBottom}
-              onChange={updateDescriptionBottom}
-            />
-            <FormControlWithLabel
-              componentClass="textarea"
-              id="media-description-side"
-              type="rich-text"
-              label={quotePh}
-              value={descriptionSide}
-              onChange={updateDescriptionSide}
-            />
-            <FormControlWithLabel
-              id="media-link"
-              type="text"
-              label={mediaLinkPh}
-              value={htmlCode}
-              onChange={(e) => {
-                return updateHtmlCode(e.target.value);
-              }}
-            />
-            <div className="admin-help">
-              <Translate value="administration.mediaHelp" />
+class MediaForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onFileSubmit = this.onFileSubmit.bind(this);
+  }
+  onFileSubmit(file) {
+    const { uploadDocument, updateHtmlCode } = this.props;
+    uploadDocument({ variables: { file: file } }).then((res) => {
+      updateHtmlCode(res.data.uploadDocument.document.externalUrl);
+    });
+  }
+  render() {
+    const {
+      descriptionTop,
+      descriptionBottom,
+      descriptionSide,
+      hasMedia,
+      htmlCode,
+      selectedLocale,
+      title,
+      toggle,
+      updateDescriptionTop,
+      updateDescriptionBottom,
+      updateDescriptionSide,
+      updateTitle,
+      updateHtmlCode
+    } = this.props;
+    const titlePh = `${I18n.t('administration.ph.title')} ${selectedLocale.toUpperCase()}`;
```

Comment:
For readability and DRY you can put selectedLocale.toUpperCase() in a variable as you use it several times